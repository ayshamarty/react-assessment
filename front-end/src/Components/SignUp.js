import React, { Component } from "react";
import axios from "axios";

import "../App.css";

class SignUp extends Component {
  constructor() {
    super();
    this.state = {
      username: "",
      email: [],
      password: "",
      password2: ""
    };
  }

  createUser = e => {
    e.preventDefault();
    const newUser = {
      username: e.target[0].value,
      email: e.target[1].value,
      password: e.target[2].value,
      password2: e.target[3].value
    };

    let notifier = document.getElementById("notifier");

    axios
      .post("http://localhost:5000/user/createuser", newUser)
      .then(response => {
        console.log(response.data);
        this.setState({
          error: response.data
        });

        if (
          response.data.email !== newUser.email &&
          response.data.password === undefined
        ) {
          notifier.style.color = "red";
          notifier.innerText = response.data.email;
        } else if (response.data.password !== newUser.password) {
          notifier.style.color = "red";
          notifier.innerText = response.data.password;
        } else if (
          response.data.email !== newUser.email &&
          response.data.password !== newUser.password
        ) {
          notifier.style.color = "red";
          notifier.innerText =
            response.data.email + "\n" + response.data.password;
        } else {
          notifier.style.color = "black";
          notifier.innerText = "user added";
        }
      })
      .catch(err => {
        notifier.innerText = err;
      });
  };

  render() {
    return (
      <div className="container">
        <form onSubmit={this.createUser}>
          <h3>Please enter your details to sign up</h3>
          <p className="asterisk">fields marked with * are compulsory</p>
          <br />
          <p>username*</p>
          <input id="create-username" />
          <br />
          <p>email address*</p>
          <input id="create-email" />
          <br />
          <p>password*</p>
          <input type="password" id="create-password" />
          <br />
          <p>verify password*</p>
          <input type="password" id="verify-password" />
          <button type="submit">Sign Up</button>
        </form>
        <div>
          <span id="notifier" />
        </div>
      </div>
    );
  }
}

export default SignUp;
