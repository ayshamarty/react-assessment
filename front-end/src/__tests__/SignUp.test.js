import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import SignUp from "../Components/SignUp";

import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<SignUp />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders shallow without crashing", () => {
  shallow(<SignUp />);
});

it('snapshot SignUp', () => {
  const wrapper = renderer.create(<SignUp/>).toJSON()
  expect(wrapper).toMatchSnapshot();
})

