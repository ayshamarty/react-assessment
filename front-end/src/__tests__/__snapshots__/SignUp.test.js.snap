// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot SignUp 1`] = `
<div
  className="container"
>
  <form
    onSubmit={[Function]}
  >
    <h3>
      Please enter your details to sign up
    </h3>
    <p
      className="asterisk"
    >
      fields marked with * are compulsory
    </p>
    <br />
    <p>
      username*
    </p>
    <input
      id="create-username"
    />
    <br />
    <p>
      email address*
    </p>
    <input
      id="create-email"
    />
    <br />
    <p>
      password*
    </p>
    <input
      id="create-password"
      type="password"
    />
    <br />
    <p>
      verify password*
    </p>
    <input
      id="verify-password"
      type="password"
    />
    <button
      type="submit"
    >
      Sign Up
    </button>
  </form>
  <div>
    <span
      id="notifier"
    />
  </div>
</div>
`;
